### ISSUE TITLE
War Thunder crashes with OpenGL injection - possible way of debugging it?

### ISSUE THREAD

**Issue by tealmill** [MaartenBaert/ssr#386]
I have been using SSR for recording War Thunder to great effect with OpenGL injection, recording at 50FPS (capped by SSR), encoding with libx264. up until some point (which I sadly can't tell when it exactly was, might have been an Nvidia driver update, might have been a game update, might have been a SSR update) when the game started crashing shortly after recording started (not launch)(SSR itself keeps running, too). However, some days, I still go and manage to record minutes of footage, on and off, without a single crash, but if it crashes once, I know it will keep crashing over and over and over.

It seems to crash after a quite random delay, sometimes it's within a single second of recording start, sometimes it's within ten seconds of it, but never more than half a minute or so. The issue is that SSR nor the game (which has built in error reporting system normally) write a single meaningful error message or anything that might give me a clue for what might be causing this (SSR just mentions the stream died). The injector is stable with other games (well, I tried only glxgears) and the SSR is stable in general.

I understand that the OpenGL injection recording mode is experimental, but this does work sometimes. There must be a reason why it happens and a way of avoiding it. Is there any way to debug the injector to get at least some chance of reading something from a traceback or such? Is there any tips for what might be causing somewhat random crashes like that?

**Comment by MaartenBaert** [comment]
You need to attach a debugger to the game that is crashing. Normally this is done by running:
```
gdb <command>
run <arguments>
```
When the crash happens, type ```t a a bt full``` to get a backtrace. This should at least tell me where the game is crashing, but unfortunately not why.

AFAIK War Thunder doesn't have a Linux version. Are you running it through WINE? Are you using Steam, or is it a stand-alone program? This can complicate debugging.

**Comment by tealmill** [comment]
No, War Thunder does have native Linux client for a good while and I am running that.

Now, though there's a major update coming very soon (possibly tomorrow) that's also meant to make considerable changes to the renderer so I would postpone debugging efforts to after that happens, in case the crash persists.

**Comment by tealmill** [comment]
Alright, it's still happening with WT 1.53. If I understood your instructions correctly, this is what you are asking for (just to be sure I am doing it right, it's the entire console log including the way I launch the debugger).

http://sinus.cz/~milan/ssr-glinject_WarThunder_crash.txt

**Comment by Tlewis334** [comment]
What command do you actually put into ssr to get it to start war thunder though?
